content/languages/c/authoring/memory-management-techniques.md

```diff
@@ -0,0 +1,24 @@
+---
+kind: tutorial
+languages: [c]
+sidebar: "language:c"
+---
+
+# Memory Management in C kata
+
+_TBD_
+
+## Arrays and strings
+
+- malloc in solution and free in tests
+- pass in a preallocated buffer (use size hints if possible)
+- two functions: get size, allocate in tests, run solution
```

Comment:
Is this approach used anywhere? The function can return the size when it gets NULL as the buffer, so how is two function better?

Reply:
This part is worked on in PR #231 which is still in draft, but you can review what's available there, would really appreciate.


> Is this approach used anywhere?

Yes, I think I've encountered it in some libraries, but my C got really rusty so I might remember some things not as they really were :)

> The function can return the size when it gets NULL as the buffer, so how is two function better?

And what if interface of the solution function has no good place to return the size? Should it use `size_t*` ans an in-out param and fill it when invoked with NULL, and read when called with non-null?